internal/zinc-apiinfo/src/main/scala/sbt/internal/inc/ClassToAPI.scala

```diff
@@ -548,56 +548,18 @@ object ClassToAPI {
   }.toMap
   def primitive(name: String): api.Type = PrimitiveRefs(name)
 
-  // Workarounds for https://github.com/sbt/sbt/issues/1035
-  //   these catch the GenericSignatureFormatError and return the erased type
-
-  private[this] def returnType(f: Field): Type =
-    try f.getGenericType
-    catch {
-      case _: GenericSignatureFormatError => f.getType
-    }
-  private[this] def parameterTypes(c: Constructor[_]): Array[Type] =
-    try c.getGenericParameterTypes
-    catch {
-      case _: GenericSignatureFormatError => convert(c.getParameterTypes)
-    }
-  private[this] def exceptionTypes(c: Constructor[_]): Array[Type] =
-    try c.getGenericExceptionTypes
-    catch {
-      case _: GenericSignatureFormatError => convert(c.getExceptionTypes)
-    }
-  private[this] def parameterTypes(m: Method): Array[Type] =
-    try m.getGenericParameterTypes
-    catch {
-      case _: GenericSignatureFormatError => convert(m.getParameterTypes)
-    }
-  private[this] def returnType(m: Method): Type =
-    try m.getGenericReturnType
-    catch {
-      case _: GenericSignatureFormatError => m.getReturnType
-    }
-  private[this] def exceptionTypes(m: Method): Array[Type] =
-    try m.getGenericExceptionTypes
-    catch {
-      case _: GenericSignatureFormatError => convert(m.getExceptionTypes)
-    }
-
-  private[this] def typeParameterTypes[T](m: Constructor[T]): Array[TypeVariable[Constructor[T]]] =
-    try m.getTypeParameters
-    catch {
-      case _: GenericSignatureFormatError => new Array(0)
-    }
-  private[this] def typeParameterTypes[T](m: Class[T]): Array[TypeVariable[Class[T]]] =
-    try m.getTypeParameters
-    catch {
-      case _: GenericSignatureFormatError => new Array(0)
-    }
-  private[this] def typeParameterTypes(m: Method): Array[TypeVariable[Method]] =
-    try m.getTypeParameters
-    catch {
-      case _: GenericSignatureFormatError => new Array(0)
-    }
-
-  private[this] def convert(classes: Array[Class[_]]): Array[Type] =
-    classes.asInstanceOf[Array[Type]] // ok: treat Arrays as read-only
+  @inline private[this] def returnType(f: Field): Type = f.getGenericType
```

Comment:
Yes, it does. Inlining is extremely important and it can cause slowdowns, but in Scala code it generally yields benefit. For example, using `@inline` in certain places of the Scala compiler are critical for performance (an example is all the statistics infrastructure). JVM inlining is normally not enough for Scala code because it will only inline up to 135 bytes, IIRC, and this is why the careful use of `@inline` is important. I saw a tweet by Jason some weeks ago that increasing this bytecode limit in the JDK helps performance for Scalac.
  
So, coming back to your question: yes, `@inline` has a real-world impact. It will not make the program 20% faster in this scenario, but it's a good practice to use `@inline` where applicable, and I think in this case it is.